lib/lhm/trigger_switcher.rb

```diff
@@ -0,0 +1,34 @@
+module Lhm
+  class TriggerSwitcher
+    attr_reader :source_table, :destination_table_name, :connection
+
+    def initialize(source_table, destination_table_name, connection)
+      @source_table = source_table
+      @destination_table_name = destination_table_name
+      @connection = connection
+    end
+
+    def copy_triggers
+      return unless !destination_table_name.empty? && connection.table_exists?(destination_table_name)
+
+      source_table.triggers.each do |trigger|
+        trigger_name = trigger[0]
+        trigger_definition = fetch_trigger_definition(trigger_name, connection)
+        # since source table is renamed to lhma_table_name(destination table name) after migration, triggers associated with the source table will now be associated with lhma_table_name.
+        # skip if the trigger isn't associated with lhma_table_name.
+        next unless trigger_definition.include? destination_table_name
+
+        modified_definition = trigger_definition.gsub(destination_table_name, source_table.name)
+        connection.execute("DROP TRIGGER #{trigger_name};")
```

Comment:
Trigger would not run for the records being changes in between the drop trigger and create trigger, 
Can we try lock tables so that records insert/updates are blocked until triggers are recreated?

Reply:
as per the updated approach, this issue will not exist